### ISSUE TITLE
Some files are wrong groups detection by import/order

### ISSUE THREAD

**Issue by mantou132** [import-js/eslint-plugin-import#2300]
expected: 
<img width="797" alt="Screen Shot 2021-11-11 at 3 40 35 PM" src="https://user-images.githubusercontent.com/3841872/141258014-8a6d90d9-b8eb-46f9-95d7-f4fca9104ab7.png">

wrong groups:
<img width="627" alt="Screen Shot 2021-11-11 at 3 40 47 PM" src="https://user-images.githubusercontent.com/3841872/141258009-7de61606-a567-4171-857d-4a90d30dde88.png">

eslint:
```js
module.exports = {
  parser: '@typescript-eslint/parser',
  extends: [
    'plugin:import/errors',
    'plugin:import/warnings',
    'plugin:import/typescript',
    'plugin:@typescript-eslint/recommended',
    'plugin:prettier/recommended',
  ],
  parserOptions: {
    ecmaVersion: 2018,
    sourceType: 'module',
  },
  settings: {
    'import/resolver': {
      typescript: {
        project: 'packages/*/tsconfig.json',
      },
    },
  },
  rules: {
    'no-console': 1,
    'import/no-named-as-default': 0,
    // https://github.com/benmosher/eslint-plugin-import/blob/HEAD/docs/rules/order.md
    'import/order': ['error', { 'newlines-between': 'always' }],
    'import/newline-after-import': 'error',
    '@typescript-eslint/no-non-null-assertion': 'off',
    '@typescript-eslint/no-var-requires': 'off',
    '@typescript-eslint/explicit-member-accessibility': 'off',
    '@typescript-eslint/explicit-function-return-type': 'off',
    '@typescript-eslint/explicit-module-boundary-types': 'off',
    '@typescript-eslint/no-explicit-any': 'off',
    '@typescript-eslint/no-unused-vars': [
      process.env.NODE_ENV === 'production' ? 2 : 1,
      {
        vars: 'all',
        args: 'after-used',
        ignoreRestSiblings: true,
        caughtErrors: 'none',
        varsIgnorePattern: '^_',
        argsIgnorePattern: '^_',
      },
    ],
  },
};
```

tsconfig:
```ts
...
"paths": {
      "src/*": ["src/*"]
    }
...
```

deps:
```
"devDependencies": {
    "@types/node": "^15.6.1",
    "@typescript-eslint/eslint-plugin": "^4.25.0",
    "@typescript-eslint/parser": "^4.25.0",
    "eslint": "^7.27.0",
    "eslint-config-prettier": "^8.3.0",
    "eslint-import-resolver-typescript": "^2.4.0",
    "eslint-plugin-import": "^2.23.4",
    "eslint-plugin-prettier": "^3.4.0",
    "husky": "^7.0.1",
    "lerna": "^4.0.0",
    "lint-staged": "^11.0.0",
    "prettier": "^2.3.0",
    "typescript": "^4.3.2"
  },
```

**Comment by ljharb** [comment]
You haven't configured the `order` rule to describe what group `src` comes from. That it's in tsconfig paths allows it to resolve, but the plugin still has no idea what kind of thing you're treating it as.
